torch/_inductor/ir.py

```diff
@@ -1738,6 +1740,83 @@ class SplitScan(Scan):
     pass
 
 
+@dataclasses.dataclass
+class ArgScan(Scan):
```

Comment:
Instead of hardcoding an `ArgScan` can you refactor `Scan` to accept multiple arguments and have `cummax` and `cummin` pass two inner fns, one of which is the index.

Reply:
I'm not sure how to use `rindex` without special casing.